beancount-core/src/flags.rs

```diff
@@ -1,12 +1,27 @@
+use std::borrow::Cow;
+
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub enum Flag {
+pub enum Flag<'a> {
     Okay,
     Warning,
-    Other(String),
+    Other(Cow<'a, str>),
 }
 
-impl Default for Flag {
+impl Default for Flag<'_> {
     fn default() -> Self {
         Flag::Okay
     }
 }
+
+impl<'a, I: Into<Cow<'a, str>>> From<I> for Flag<'a> {
```

Comment:
You are using `Into` here unlike in `From<Cow<_>> for Date`. This think those should be consistent. Which is better?

Reply:
I think `From` is preferred when possible (it's not always possible), since implementing `From` gets you `Into` for free.

https://doc.rust-lang.org/stable/rust-by-example/conversion/from_into.html

Reply:
What I'm trying to point out is that the `From` impl for `Flag` is generic over `I: Into<Cow<'a, str>>` while the `From` impl for `Date` is not:
```rust
 impl<'a, I: Into<Cow<'a, str>>> From<I> for Flag<'a> {}
 impl<'a> From<Cow<'a, str>> for Date<'a> {}
```
I think these should be consistent since they are both wrappers around strings.

Reply:
That's true. I think that it's better to drop the generics, and potentially add more implementations for `&str` and `String`. The Rust compiler had trouble deciding which implementation to use, but I'm not sure why, as `NaiveDate` doesn't implement `Into<Cow>`.

It doesn't matter much whether to use `Into` or `From`, it's just that when implementing `From` is preferable. In any case, a type has either none,  just `Into` or both.